pkg/controllers/clusterresourceplacement/controller_integration_test.go

```diff
@@ -1868,26 +1888,90 @@ var _ = Describe("Test ClusterResourcePlacement Controller", func() {
 						{Name: ptr.To("generation"), Value: ptr.To(strconv.FormatInt(gotCRP.Generation, 10))},
 						{Name: ptr.To("conditionType"), Value: ptr.To("Completed")},
 						{Name: ptr.To("status"), Value: ptr.To(string(corev1.ConditionTrue))},
+						{Name: ptr.To("reason"), Value: ptr.To("CRPCompleted")},
 					},
 					Gauge: &prometheusclientmodel.Gauge{
 						Value: ptr.To(float64(time.Now().UnixNano()) / 1e9),
 					},
 				},
 			)
-			checkPlacementStatusMetric(customRegistry, wantMetrics)
+			checkPlacementStatusMetric(wantMetrics)
 		})
 	})
-})
 
-func registerMetrics() {
-	// Create a test registry
-	customRegistry = prometheus.NewRegistry()
-	Expect(customRegistry.Register(metrics.FleetPlacementStatusLastTimeStampSeconds)).Should(Succeed())
-	metrics.FleetPlacementStatusLastTimeStampSeconds.Reset()
-}
+	Context("When creating an ClusterResourcePlacement with user error", func() {
```

Comment:
Can we add a simple test case for CRP with external rollout strategy? No need to create an updateRun, it should be in RolloutStartedUnknown state with reason "RolloutControlledByExternalController". Thank you!